### PR TITLE
[PKG-2941] langchain 0.0.298 :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| sfe1ed40 | [Conda Downloads](https://anaconda.org/sfe1ed40/langchain) | [Conda Version](https://anaconda.org/sfe1ed40/langchain) | [Conda Platforms](https://anaconda.org/sfe1ed40/langchain) | Conda License |

Changelog: https://github.com/langchain-ai/langchain/releases
License: https://github.com/langchain-ai/langchain/blob/v0.0.298/LICENSE
Requirements: https://github.com/langchain-ai/langchain/blob/v0.0.298/libs/langchain/pyproject.toml

Actions:
1. Add `abs.yaml `with `channels` to get a dependency `langsmith`
2. Add the `github-source` folder from the github url to copy a missing `LICENSE` file
3. Add to script: `cp github-source/LICENSE LICENSE `
4. Remove `wheel` from `host` as poetry has its own wheel builder
5. Update runtime dependencies: update pinnings, add `langsmith`, move `tqdm` to `run_constrained`
6. Update `home` and `doc_url`

Notes:
- **To be added**: `upload_channels: sfe1ed40` before merging

### CODE REVIEW

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by cbouss — COMMENTED** [review]
Except run_constrained, the rest looks good.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
-    - openapi-schema-pydantic >=1.2.0,<2.0.0
-    - dataclasses-json >=0.5.7,<0.6.0
+    - dataclasses-json >=0.5.7,<0.7.0
     - tenacity >=8.1.0,<9.0.0
     - aiohttp >=3.8.3,<4.0.0
     - async-timeout >=4.0.0,<5.0.0  # [py<311]
     - numexpr >=2.8.4,<3.0.0
-    - tqdm >=4.48.0
+    - langsmith >=0.0.38,<0.1.0
   run_constrained:
```
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/meta.yaml:42]: Thanks, I've updated it

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,79 +1,84 @@
...
+    sha256: d8425a05aee5dc5474d2a1fae766b7e19b5b08d8facac2d4a81353ce3df80322
+  - url: https://github.com/langchain-ai/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: b24039599f2ba23047b1aacaeb656361854d813d4d35f6088d2a7cfb7f809f6b
+    folder: github-source
 
 build:
+  number: 0
   skip: True # [py<38]
+  script: 
+    - cp github-source/LICENSE LICENSE
```
Comment (boldorider4) [inline recipe/meta.yaml:19]: this should be done in the about section I think. Unless it's the unpacking of the second tarball that causes issues.

Comment (skupr-anaconda) [reply recipe/meta.yaml:19]: The license is missing in the source dist on PyPi. Conda-forge for some reason also has 2 source/url

Comment (boldorider4) [reply recipe/meta.yaml:19]: So that license file is coming from the second tarball, is that right?

Comment (skupr-anaconda) [reply recipe/meta.yaml:19]: Yep

**Review by boldorider4 — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
Looks Good

**Pull request merged**